src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -6122,6 +6114,750 @@ .maxstack  3
                 """);
         }
 
+        [CombinatorialData]
+        [Theory]
+        public void SynthesizedReadOnlyArray([CombinatorialValues("IEnumerable<object>", "IReadOnlyCollection<object>", "IReadOnlyList<object>")] string targetType)
+        {
+            string source = $$"""
+                using System;
+                using System.Collections;
+                using System.Collections.Generic;
+                using static System.Console;
+                class Program
+                {
+                    static void Main()
+                    {
+                        Report([1, 2, null]);
+                    }
+                    static void Report({{targetType}} x)
+                    {
+                        Write("IEnumerable.GetEnumerator(): ");
+                        ((IEnumerable)x).Report(includeType: true);
+                        WriteLine();
+                        Write("IEnumerable<object>.GetEnumerator(): ");
+                        ((IEnumerable<object>)x).Report(includeType: true);
+                        WriteLine();
+                        WriteLine("IReadOnlyCollection<object>.Count: {0}", ((IReadOnlyCollection<object>)x).Count);
+                        WriteLine("IReadOnlyList<object>.this[]: {0}", ((IReadOnlyList<object>)x)[1]);
+                        WriteLine("ICollection<object>.Count: {0}", ((ICollection<object>)x).Count);
+                        WriteLine("ICollection<object>.IsReadOnly: {0}", ((ICollection<object>)x).IsReadOnly);
+                        WriteLine("ICollection<object>.Add: {0}", Invoke(() => ((ICollection<object>)x).Add(-1)));
+                        WriteLine("ICollection<object>.Clear: {0}", Invoke(() => ((ICollection<object>)x).Clear()));
+                        WriteLine("ICollection<object>.Contains: {0}", ((ICollection<object>)x).Contains(2));
```

Comment:
Same comment here, basically whenever the argument affects the behavior we expect. It's a small nit so NBD if you decide not to fix in this PR.